docs/odl9-release-planning.md

```diff
@@ -0,0 +1,91 @@
+# ODL 9 Release Planning
+
+## Project Goal
+The objective of ODL 9 is to enhance the OData .NET Libraries by addressing significant issues and implementing key features identified as major release opportunities.
+
+This release aims to improve performance, security, and usability, ensuring the libraries meet the evolving needs of developers and organizations. The General Availability (GA) release is scheduled for November 2025, with multiple preview, beta, and release candidate versions planned to gather feedback and ensure stability.
+
+## Scope
+The scope of ODL 9 includes evaluating and potentially incorporating the issues in the table below. Each item will be assessed based on its impact, feasibility, and alignment with the project's goals.
+
+## .NET Framework Support
+ODL 9 will align with modern .NET platform support to ensure compatibility and long-term maintainability. The release will support:
+- **.NET 8 (LTS)** – Ensuring support for the latest long-term support version – End of support: Nov 10, 2026
+- **.NET 9 (STS)** – Ensuring support for the latest standard-term support version – End of support: May 12, 2026
+- **.NET 10 (LTS) Preview?** – Preview expected in Feb 2025, General Availability (GA) expected on Nov 11, 2025
+
+## Proposed Items for Inclusion
+| Item | Details | Library | Status | Remark |
+| ---- | ----------- | ------- | ------ | ------ |
+| **[2908](https://github.com/OData/odata.net/issues/2908): Remove `OdataMessageInfo` from the DI container** | It is pointless to have this injectable as a service since we overwrite all the properties of the resolved instance. | Core | In scope | |
+| **[2907](https://github.com/OData/odata.net/issues/2907): Remove `IContainerProvider` and pass the `IServiceProvider` directly to the message reader or writer** |  |  Core | In scope | Consult with Robert McLaws to gather his input on how best to redesign dependency injection in the writers and readers |

```

Comment:
I agree with Robert's comments in the github issue. That said, even if we can't overhaul the DI patterns that we use, I think this particular area is a massive step in the right direction for improved DI practices. So, I think we should do this work even if we find it's out of scope to overhaul all of our DI usage.